ca/certificate-authority.go

```diff
@@ -362,18 +411,55 @@ func (ca *CertificateAuthorityImpl) IssueCertificate(csr x509.CertificateRequest
 		ca.log.Notice(message)
 	}
 
-	// Convert the CSR to PEM
-	csrPEM := string(pem.EncodeToMemory(&pem.Block{
-		Type:  "CERTIFICATE REQUEST",
-		Bytes: csr.Raw,
-	}))
+	// At this point, we are willing to issue, so create a pending
+	// certificate request.
+	pendingCert := core.Certificate{RegistrationID: regID}
+	pendingCert, err = ca.SA.NewPendingCertificate(pendingCert)
+	if err != nil {
+		ca.log.AuditErr(err)
+		return emptyCert, err
+	}
+
+	// Dispatch signing request, or fail if the queue is full
+	request := signingRequest{
+		logEventID:     logEventID,
+		cert:           pendingCert,
+		csr:            csr,
+		commonName:     commonName,
+		hostNames:      hostNames,
+		earliestExpiry: earliestExpiry,
+	}
+	select {
```

Comment:
While this wouldn't be necessarily *completely* async, it might be nice to block on adding the request to the channel for X seconds so that if the queue is full but is about to have a slot free up we don't immediately return an error..?